src/HealthChecks.Aws.S3/S3BucketOptions.cs

```diff
@@ -1,12 +1,17 @@
 using Amazon.S3;
 using Amazon.S3.Model;
+using Amazon.Runtime;
+using System;
 
 namespace HealthChecks.Aws.S3
 {
     public class S3BucketOptions
     {
-        public string? AccessKey { get; set; }
+        public AWSCredentials Credentials { get; set; } = null!;
 
+        [Obsolete("Specify access key and secret as a BasicCredential to AWSCredentials instead")]
+        public string? AccessKey { get; set; }
+        [Obsolete("Specify access key and secret as a BasicCredential to AWSCredentials instead")]
         public string? SecretKey { get; set; }
 
         public AmazonS3Config S3Config { get; set; } = null!;
```

Comment:
```suggestion
        public AmazonS3Config? S3Config { get; set; }
```

Reply:
I would prefer to not change anything not already impacted by my changes. So I would suggest doing this change as a seperate PR.

Reply:
1-line non-breaking change. Whatever.